### PR TITLE
sanoid/syncoid: Add package options

### PR DESCRIPTION
**Pull request by randomnetcat**
Allows changing the sanoid package to be used for these services.

Maintainers: @julm @lopsided98 

###### Description of changes

Add package options to `sanoid` and `syncoid` modules.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by jian-lin — APPROVED** [review]

File: nixos/modules/services/backup/sanoid.nix [hunk 1]
```diff
@@ -114,6 +114,15 @@ in
   options.services.sanoid = {
     enable = mkEnableOption (lib.mdDoc "Sanoid ZFS snapshotting service");
 
+    package = mkOption {
```
[the remark this answers is not in the document]
Comment (randomnetcat) [reply nixos/modules/services/backup/sanoid.nix]: Updated, thanks.

File: nixos/modules/services/backup/syncoid.nix [hunk 2]
```diff
@@ -87,6 +87,8 @@ in
   options.services.syncoid = {
     enable = mkEnableOption (lib.mdDoc "Syncoid ZFS synchronization service");
 
+    package = lib.mkPackageOption pkgs "sanoid" {};
```
Comment (SuperSandro2000) [inline nixos/modules/services/backup/syncoid.nix:90]:
```suggestion
    package = lib.mkPackageOptionMD pkgs "sanoid" {};
```

File: nixos/modules/services/backup/sanoid.nix [hunk 3]
```diff
@@ -114,6 +114,8 @@ in
   options.services.sanoid = {
     enable = mkEnableOption (lib.mdDoc "Sanoid ZFS snapshotting service");
 
+    package = lib.mkPackageOption pkgs "sanoid" {};
```
Comment (SuperSandro2000) [inline nixos/modules/services/backup/sanoid.nix:117]:
```suggestion
    package = lib.mkPackageOptionMD pkgs "sanoid" {};
```

**Comment by Kranzes** [comment]
Please squash the commits so we have a clean history.

**Comment by randomnetcat** [comment]
Done.

**Pull request merged**